### PR TITLE
Moved GetMojo logic to an abstract class. This allows new mojos to im…

### PR DESCRIPTION
**Pull request by jonvolfson**
…plement functionality on top of checking if an artifact can be retrieved remotely.

This will allow the logic in the old GetMojo to be used for [MDEP-645].

### CODE REVIEW

**Comment by jonvolfson** [comment]
@bimargulies @elharo

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+import org.apache.maven.shared.transfer.dependencies.resolve.DependencyResolver;
+import org.apache.maven.shared.transfer.dependencies.resolve.DependencyResolverException;
+import org.codehaus.plexus.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
```
Comment (bimargulies-google) [inline src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java:53]: Add javadoc comment here that explains what this class is good for.

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+            throws MojoExecutionException, MojoFailureException;
+
+    @Override
+    public void execute()
+            throws MojoExecutionException, MojoFailureException
+    {
+        doExecute();
+    }
+
+    public ProjectBuildingRequest buildBuildingRequest()
```
Comment (bimargulies-google) [inline src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java:170]: Need javadoc comment on here, should explain that the expectation is that derived classes will call this.

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+    /**
+     * Skip plugin execution completely.
+     *
+     * @since 2.7
+     */
+    @Parameter( property = "mdep.skip", defaultValue = "false" )
+    private boolean skip;
+
+    /**
+     * @throws MojoExecutionException {@link MojoExecutionException}
```
Comment (bimargulies-google) [inline src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java:157]: Needs text stating that each derived class will provide this.

File: src/main/java/org/apache/maven/plugins/dependency/GetMojo.java [hunk 4]
```diff
@@ -58,292 +58,38 @@
...
-    private boolean skip;
 
     @Override
-    public void execute()
-        throws MojoExecutionException, MojoFailureException
+    protected void doExecute()
+            throws MojoExecutionException, MojoFailureException
     {
+
         if ( isSkip() )
```
Comment (bimargulies-google) [inline src/main/java/org/apache/maven/plugins/dependency/GetMojo.java:70]: Skip logic can be in base class.

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java (hunk 1 above)
Comment (jonvolfson) [reply src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java:53]: Resolved

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java (hunk 2 above)
Comment (jonvolfson) [reply src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java:182]: Resolved

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java (hunk 3 above)
Comment (jonvolfson) [reply src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java:160]: Resolved

File: src/main/java/org/apache/maven/plugins/dependency/GetMojo.java (hunk 4 above)
Comment (jonvolfson) [reply src/main/java/org/apache/maven/plugins/dependency/GetMojo.java]: Resolved

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+ */
+public abstract class AbstractGetMojo
+        extends AbstractMojo
+{
+    private static final Pattern ALT_REPO_SYNTAX_PATTERN = Pattern.compile( "(.+)::(.*)::(.+)" );
+
+    @Parameter( defaultValue = "${session}", required = true, readonly = true )
+    private MavenSession session;
+
+    /**
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java:65]: no empty doc comments please

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+    private List<ArtifactRepository> pomRemoteRepositories;
+
+    /**
+     * Download transitively, retrieving the specified artifact and all of its dependencies.
+     */
+    @Parameter( property = "transitive", defaultValue = "true" )
+    private boolean transitive = true;
+
+    /**
+     * Skip plugin execution completely. Derived classes will provide this.
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java:152]: skip seems to be done in this class

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java (hunk 5 above)
Comment (jonvolfson) [reply src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java]: resolved

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java (hunk 6 above)
Comment (jonvolfson) [reply src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java]: resolved

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+    private String classifier;
+
+    /**
+     * The packaging of the artifact to download. Ignored if {@link #artifact} is used.
+     */
+    @Parameter( property = "packaging", defaultValue = "jar" )
+    private String packaging = "jar";
+
+    /**
+     * Repositories in the format id::[layout]::url or just url, separated by comma. ie.
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java:121]: ie. --> For example,

File: src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Abstract parent class used by mojos that retrieve artifacts from the remote repository.
+ */
+public abstract class AbstractGetMojo
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/dependency/AbstractGetMojo.java:57]: Is there maybe a better name for this class? GetMojo implies this is for the dependency:get goal.

**Pull request closed without merging**